testutils/assertions.go

```diff
@@ -6,23 +6,21 @@ import (
 )
 
 // ExpectEqualProtoMessages provides richer error messages than struct comparison by leveraging the String() method that all
-// proto Messages provide. On error, Gomega's string comparison utility prints a few characters of the text immediately
-// surrounding the first discrepancy.
+// proto Messages provide.
 //
 // Variadic optionalDescription argument is passed on to fmt.Sprintf() and is used to annotate failure messages.
 //
 // Example of the output:
 //   optionalDescription is rendered here: template string with values foo and bar
 //   Expected
-//       <string>: "...-1010" vers..."
-//   to equal               |
-//       <string>: "...-10101" ver..."
+//     <string>: application_state:<installation_state:<install_namespace:"linkerd-ns" > status:<> metadata:<> registry_name:"hub-meshes" application_name:"linkerd2" >
+//   to be equivalent to
+//     <string>: mesh:<status:<> metadata:<name:"istio-mesh" namespace:"test-install" resource_version:"2" > istio:<installation_namespace:"istio-system" version:"1.0.6" > >
 func ExpectEqualProtoMessages(a, b proto.Message, optionalDescription ...interface{}) {
 	if proto.Equal(a, b) {
 		return
 	}
-	// One shortcoming is that you only get +/- 5 chars of context
-	// per: https://github.com/onsi/gomega/blob/master/format/format.go#L146
-	// TODO(mitchdraft) gomega pr to modify charactersAroundMismatchToInclude (if not merged will make a util)
-	Expect(a.String()).To(Equal(b.String()), optionalDescription...)
+	// Prefer BeEquivalentTo over Equal because it prints entire strings for
+	// the actual and expected values.
+	Expect(a.String()).To(BeEquivalentTo(b.String()), optionalDescription...)
```

Comment:
now that the gomega pr merged you need to bump gomega to v1.7.0 and set `CharactersAroundMismatchToInclude` to a large number
might want to do
```
initalCharactersToInclude := CharactersAroundMismatchToInclude
CharactersAroundMismatchToInclude = 10000
```
before the `Expect`, and:
`CharactersAroundMismatchToInclude = initialCharactersToInclude`
after the `Expect`

Reply:
oh I see what you are doing, how about this:
```
if a.String() != b.String() {
  fmt.Fprintf("protos do not match\nhave (length %v):\n%v\n want (length %v):\n %v", len(a.String()),a.String(),len(b.String()),b.String())
  fmt.Fprintf(GinkgoWriter, "set CharactersAroundMismatchToInclude to include more or less diff context (current value is )", CharactersAroundMismatchToInclude)
}
Expect(a.String()).To(Equal(b.String()), optionalDescription...)
```

i.e. use both approaches

may as well be generous with the clues on this central assertion util